### PR TITLE
Added bit bucket backend

### PR DESCRIPTION
**Pull request by vivekanand1101**
Presently, i have removed 'tip' from the release list as it doesn't appear to be one of the releases.

Fixes https://github.com/fedora-infra/anitya/issues/141

### CODE REVIEW

File: anitya/lib/backends/bitbucket.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+
+"""
+ (c) 2014 - Copyright Red Hat Inc
```
Comment (pypingou) [inline anitya/lib/backends/bitbucket.py]: That's incorrect :)

File: anitya/lib/backends/bitbucket.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+ Authors:
+   Pierre-Yves Chibon <pingou@pingoured.fr>
```
Comment (pypingou) [inline anitya/lib/backends/bitbucket.py]: This as well :)

File: anitya/lib/backends/bitbucket.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            corresponds to the current plugin.
+        :return: a list of all the possible releases found
+        :return type: list
+        :raise AnityaPluginException: a
+            :class:`anitya.lib.exceptions.AnityaPluginException` exception
+            when the versions cannot be retrieved correctly
+
+        '''
+        if project.version_url:
+            url_template = 'https://bitbucket.org/%(version_url)s/downloads'
```
Comment (pypingou) [inline anitya/lib/backends/bitbucket.py]: Should this be the version_url or the project name?

**Comment by pypingou** [comment]
So couple of question but the big missing point is the unit-tests :)

**Comment by ralphbean** [comment]
> the big missing point is the unit-tests

Agreed.  Anitya has a good test suite already, it would be good to keep parity between new modules and new tests.

@vivekanand1101, do you know where the tests are and how to run them?

**Comment by pypingou** [comment]
@vivekanand1101 said on IRC that he would not have time to fix this PR before a while, so I went ahead and finished it for him, the review is available at: https://github.com/fedora-infra/anitya/pull/218

**Pull request closed without merging**